test/hotspot/jtreg/serviceability/jvmti/vthread/ToggleNotifyJvmtiTest/ToggleNotifyJvmtiTest.java

```diff
@@ -0,0 +1,178 @@
+            if (n <= 0) {
+                n = 1000;
+                ToggleNotifyJvmtiTest.sleep(1);
+            }
+            if (i > n) {
+                i = 0;
+                n = n - 1;
+            }
+            i = i + 1;
+        }
+    }
+
+    // ensure thread is ready
+    public void ensureReady() {
+        try {
+            while (!threadReady) {
+                sleep(1);
+            }
+        } catch (InterruptedException e) {
+            throw new RuntimeException("Interruption while preparing tested thread: \n\t" + e);
+        }
+    }
+
+    public void letFinish() {
+        shouldFinish = true;
+    }
+}
+
+public class ToggleNotifyJvmtiTest {
+    private static final int VTHREADS_CNT = 60;
+    private static final String AGENT_LIB = "ToggleNotifyJvmtiTest";
+    private static final WhiteBox WB = WhiteBox.getWhiteBox();
+
+    private static native boolean IsAgentStarted();
+    private static native int VirtualThreadStartedCount();
+
+    static void log(String str) { System.out.println(str); }
+
+    static public void sleep(long millis) {
+        try {
+            Thread.sleep(millis);
+        } catch (InterruptedException e) {
+            throw new RuntimeException("Interruption in TestedThread.sleep: \n\t" + e);
+        }
+    }
+
+    static TestedThread[] threads = new TestedThread[VTHREADS_CNT];
+    static Thread vts[] = new Thread[VTHREADS_CNT];
+
+    static private synchronized void startThread(int i) {
+        String name = "TestedThread" + i;
+        TestedThread thread = new TestedThread(name);
+        vts[i] = Thread.ofVirtual().name(name).start(thread);
+        thread.ensureReady();
+        threads[i] = thread;
+        log("## Java: started vthread: " + name);
+    }
+
+    static private void startThreads() {
+        log("\n## Java: Starting vthreads");
+        for (int i = 0; i < VTHREADS_CNT; i++) {
+            startThread(i);
+        }
+    }
+
+    static private synchronized void finishThreads() {
+        log("\n## Java: runIt: Finishing vthreads");
+        try {
+            for (int i = 0; i < VTHREADS_CNT; i++) {
+                TestedThread thread = threads[i];
+                if (thread == null) {
+                    break;
+                }
+                thread.letFinish();
+                vts[i].join();
+            }
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    static private void setVirtualThreadsNotifyJvmtiMode(int iter, boolean enable) {
+        sleep(5);
+        WB.setVirtualThreadsNotifyJvmtiMode(enable);
+        log("# main: SetNotifyJvmtiEvents: #" + iter + " enable: " + enable);
+    }
+
+    public static void main(String[] args) throws Exception {
+        log("# main: loading " + AGENT_LIB + " lib");
+
+        if (args.length > 0 && args[0].equals("attach")) { // agent loaded into running VM case
+            String arg = args.length == 2 ? args[1] : "";
```

Comment:
I don't see any args being passed in other than "attach"? What might `arg` be set to?

Reply:
Only "attach" can be passed in args.

Reply:
So if args[0] is "attach", then arg will always be set to "attach",  which means it could be passed in as a literal to loadAgentLibrary() below. But even that doesn't seem to be necessary since the agent doesn't seem to look at the args passed in.